iwyu_output.cc

```diff
@@ -1062,13 +1062,43 @@ void ProcessForwardDeclare(OneUse* use,
   }
 }
 
+// Returns true if the given symbol has a mapping defined to a file, and that
+// file does not include the given FileEntry (and thus is safe to be included
+// by the FileEntry).
+static bool HasMappingSuitableForFile(
+    const string& symbol, const FileEntry* file,
+    const IwyuPreprocessorInfo* preprocessor_info) {
+  for (const std::string& public_file :
+       GlobalIncludePicker().GetCandidateHeadersForSymbol(symbol)) {
+    // To avoid cyclic includes we must only count files that do not
+    // include this one.
+    const FileEntry* public_file_entry =
+        preprocessor_info->IncludeToFileEntry(public_file);
+    if (public_file_entry == nullptr ||
+        !preprocessor_info->FileTransitivelyIncludes(public_file_entry, file)) {
```

Comment:
Hmm. I wonder if it's legit to use `FileTransitivelyIncludes` in the IWYU output stage. I see that it's been used before, but I remember running into problems in PR #594 because I couldn't be sure that the include I was basing decisions on wouldn't be added/removed later in the analysis. Or something like that.

So in this example, what if `public_file_entry` does not include `file`, but a later analysis pass adds it? Will that invalidate the analysis?

It does seem unlikely to me that we would be editing files we have mappings for, though... Thoughts?

Reply:
Thanks for walking me through this. Let's start with the simpler same-file check, and we can circle back to this if it becomes a problem. Don't forget to update the doc comment for the function.

Reply:
I've had an idea for an alternative solution that might be neater.  I'm going to investigate that first.